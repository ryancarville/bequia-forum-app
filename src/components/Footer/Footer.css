.footer-container {
	width: 100%;
	font-size: 16px;
	text-align: center;
	padding-top: 100px;
}
.footer-content {
	display: grid;
	grid-template-columns: repeat(5, 0.5fr);
}
.footer-content a {
	text-decoration: none;
	color: green;
	padding: 10px;
	cursor: pointer;
}

.footer-container p {
	font-size: 12px;
	padding: 10px;
}
