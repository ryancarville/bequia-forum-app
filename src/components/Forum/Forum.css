.forum-container {
  width: 100%;
  font-size: 20px;
  text-align: center;
  margin-top: 80px;
}
.forum-container h3 {
  text-align: center;
}
.forum-container button {
  width: 10%;
}
.forum-content {
  width: 100%;
  display: block;
  margin-left: auto;
  margin-right: auto;
}
.sectionMenu {
  width: 100%;
  list-style: none;
  padding: 0px;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(100px, 1fr));
}
.section-menu-link {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  background-color: blue;
  color: white;
  width: 100px;
  height: 100px;
  margin: 0;
  padding: 10px;
  font-size: 20px;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
  text-decoration: none;
  border-radius: 10px;
  animation: moveRight 1s ease-in-out;
}

@media only screen and (min-width: 700px) {
  .forum-container {
    margin-top: 20px;
  }
  .sectionMenu {
    width: 50%;
    grid-template-columns: repeat(auto-fill, minmax(125px, 1fr));
    grid-column-gap: 100px;
    margin-left: auto;
    margin-right: auto;
  }
  .section-menu-link {
    width: 200px;
    animation: none;
  }
}
