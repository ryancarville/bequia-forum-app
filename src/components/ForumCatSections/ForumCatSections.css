.forum-cat-sections-container {
	width: 100%;
	margin-top: 80px;
}
.forum-cat-sections-content {
	width: 100%;
	font-size: 20px;
	text-align: center;
}

.forum-cat-sections-content header h3 {
	margin: 0;
	margin-top: 10px;
}
.forum-cat-sections-content ul {
	list-style: none;
	margin: 0;
	padding: 0;
	display: flex;
	flex-direction: column;
	align-items: center;
}

.section-cat-item {
	width: 90%;
	height: auto;
	padding: 10px;
	margin-top: 20px;
	text-align: center;
	border: 1px solid blue;
	border-radius: 10px;
}
.section-cat-item a {
	display: flex;
	justify-content: space-between;
	align-items: center;
	color: black;
	text-decoration: none;
}
.section-text {
	width: 60%;
	text-align: left;
}
.thread-count {
	font-size: 20px;
	margin: 10px;
	width: 50px;
	display: flex;
	justify-content: space-between;
	align-items: center;
}
#wave-loader-gif {
	width: 30px;
	vertical-align: middle;
}

@media only screen and (min-width: 700px) {
	.forum-cat-sections-container {
		margin-top: 50px;
	}
	.section-cat-item {
		font-size: 16px;
		padding-top: 0px;
		padding-bottom: 0;
	}
}
