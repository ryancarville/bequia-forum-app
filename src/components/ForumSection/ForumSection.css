.forum-section-container {
  width: 100%;
  font-size: 20px;
  display: flex;
  flex-direction: column;
  margin-top: 50px;
  opacity: 0;
  animation: fadeIn 2s ease-in forwards;
}
.forum-section-container h3 {
  text-align: center;
}

.forum-section-content ul {
  width: 90%;
  padding: 0px;
  list-style: none;
  margin-left: auto;
  margin-right: auto;
}

.post-card {
  display: flex;
  align-items: center;
  flex-direction: column;
  border-bottom: 1px solid blue;
  padding: 10px;
  padding-bottom: 0;
  margin: 0;
  margin-bottom: 20px;
  height: auto;
  font-size: 18px;
  line-height: 1.5em;
  text-align: center;
}
.post-card-info {
  width: 100%;
}
.post-card-info a {
  text-decoration: none;
  margin: 0;
}
.post-card-info h4 {
  margin-top: 0;
  margin-bottom: 0px;
  font-size: 20px;
}
.post-teaser {
  display: none;
}
.post-card-user-info {
  line-height: 1.5em;
  bottom: 0;
}
.post-card-user-info p {
  margin: 0;
}
.post-card-icons {
  width: 100%;
  height: 30px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin: 0px;
  padding: 0px;
  margin-bottom: 5px;
}
.post-icons p {
  margin-right: 10px;
}

@media only screen and (min-width: 700px) {
  .forum-section-container {
    font-size: 18px;
    display: block;
    margin-top: 20px;
  }
  .forum-section-container h3 {
    margin-left: 100px;
    float: left;
  }

  .forum-section-content {
    padding: 25px;
  }
  .forum-section-content ul {
    margin-top: 90px;
  }
  .forum-loader {
    margin-top: 70px;
  }
  .error-message {
    text-align: center;
    margin-top: 100px;
  }
  .post-card {
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-start;
    text-align: left;
    padding: 15px;
    font-size: 15px;
  }
  .post-card-info {
    width: 70%;
    bottom: 0;
  }
  .post-card-info a {
    text-decoration: none;
    margin: 0;
  }
  .post-card-info h4 {
    margin-top: 0;
    margin-bottom: 0px;
    font-size: 20px;
  }
  .post-teaser {
    display: block;
  }
  .post-card-user-info {
    line-height: 1.5em;
    text-align: right;
  }
  .post-card-user-info p {
    margin: 0;
  }
  .post-card-icons {
    width: 100%;
    height: 30px;
    display: flex;
    justify-content: space-around;
    align-items: center;
    margin: 0px;
    padding: 0px;
    margin-bottom: 5px;
  }
  .post-icons p {
    margin-right: 10px;
  }
}
