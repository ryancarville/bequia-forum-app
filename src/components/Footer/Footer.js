import React from "react";
import { Link } from "react-router-dom";
import "./Footer.css";
//footer
export default function Footer(props) {
  return (
    <section className="footer-container">
      <div
        className="footer-content"
        onLoad={props.handleNavTextColor}
        onClick={props.handleNavTextColor}
      >
        <Link to={"/about-us"}>About</Link>
        <Link to={"/help"}>Support</Link>
        <Link to={"/contact-us"}>Contact</Link>
        <Link to={"/donate"}>Donate</Link>
        <Link to={"/advertising"}>Advertise</Link>
      </div>
      <br />
      <p>All Rights Reserved Â©2019 Bequia Forum</p>
    </section>
  );
}
