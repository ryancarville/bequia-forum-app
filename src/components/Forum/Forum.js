import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./Forum.css";
import waveLoader from "../Icons/waveLoader";
import apiServices from "../../services/apiServices";
//forum component
export default class Forum extends Component {
  constructor(props) {
    super(props);
    this.state = {
      forumTitles: [],
      error: null
    };
  }
  //make forum sections
  makeForum = () => {
    let links = [];
    this.state.forumTitles.forEach(title => {
      if (title.name === "Jobs") {
        links.push(
          <Link to={"/jobs"} className="section-menu-link" key={title.id}>
            {title.name}
            <i className="fas fa-briefcase" samesite="none" secure="true"></i>
          </Link>
        );
      }
      if (title.name === "Market Place") {
        links.push(
          <Link
            className="section-menu-link"
            to={"/marketPlace"}
            key={title.id}
          >
            {title.name}
            <i className="fas fa-store" samesite="none" secure="true"></i>
          </Link>
        );
      }
      if (title.name === "Rentals") {
        links.push(
          <Link to={"/rentals"} className="section-menu-link" key={title.id}>
            {title.name}
            <i className="fas fa-home" samesite="none" secure="true"></i>
          </Link>
        );
      }
      if (title.name === "Events") {
        links.push(
          <Link to={"/events"} className="section-menu-link" key={title.id}>
            {title.name}
            <i
              className="far fa-calendar-alt"
              samesite="none"
              secure="true"
            ></i>
          </Link>
        );
      }
      if (title.name === "Life on Bequia") {
        links.push(
          <Link
            to={`/messageBoard/${title.id}`}
            className="section-menu-link"
            key={title.id}
          >
            {title.name}
            <i
              className="fas fa-umbrella-beach"
              samesite="none"
              secure="true"
            ></i>
          </Link>
        );
      }
      if (title.name === "Help & Tips") {
        links.push(
          <Link
            to={`/messageBoard/${title.id}`}
            className="section-menu-link"
            key={title.id}
          >
            {title.name}
            <i
              className="fas fa-people-carry"
              samesite="none"
              secure="true"
            ></i>
          </Link>
        );
      }
      if (title.name === "Activities") {
        links.push(
          <Link
            to={`/messageBoard/${title.id}`}
            className="section-menu-link"
            key={title.id}
          >
            {title.name}
            <i className="fas fa-hiking" samesite="none" secure="true"></i>
          </Link>
        );
      }
      if (title.name === "Off-Topic") {
        links.push(
          <Link
            to={`/messageBoard/${title.id}`}
            className="section-menu-link"
            key={title.id}
          >
            {title.name}
            <i className="fas fa-question" samesite="none" secure="true"></i>
          </Link>
        );
      }
      if (title.name === "Support") {
        links.push(
          <Link
            to={`/messageBoard/${title.id}`}
            className="section-menu-link"
            key={title.id}
          >
            {title.name}
            <i className="fas fa-info" samesite="none" secure="true"></i>
          </Link>
        );
      }
    });
    return links;
  };
  componentDidMount() {
    window.scroll(0, 0);
    apiServices.getFourmSectionTitles().then(titles => {
      if (titles.error) {
        this.setState({
          error: titles.error
        });
      } else {
        this.setState({
          forumTitles: titles
        });
      }
    });
  }

  render() {
    return (
      <section className="forum-container">
        <div className="forum-content">
          <header>
            <h3>Forum</h3>
          </header>
          {this.state.error ? <p>{this.state.error}</p> : null}
          {this.state.forumTitles.length > 0 ? (
            <ul className="sectionMenu">{this.makeForum()}</ul>
          ) : (
            waveLoader
          )}
        </div>
      </section>
    );
  }
}
