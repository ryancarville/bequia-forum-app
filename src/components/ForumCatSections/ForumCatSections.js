import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./ForumCatSections.css";
import apiServices from "../../services/apiServices";
//forum section
export default class ForumCatSections extends Component {
  constructor(props) {
    super(props);
    this.state = {
      forumTitle: "",
      threadCounts: [],
      boards: [],
      error: null
    };
  }
  //get number of threads for a board
  getCount = id => {
    const num = this.state.threadCounts.filter(count => count.board_id === id);
    return num[0].count;
  };
  //make all the boards for the section
  makeBoards = () => {
    return this.state.boards.map(item => {
      return (
        <li key={item.id} className="section-cat-item">
          <Link to={`/messageBoard/${item.messageboard_section}/${item.id}`}>
            <div className="section-text">
              <h4>{item.name}</h4>
              <p>{item.description}</p>
            </div>
            <span className="thread-count">
              <h5>Threads</h5>

              <p id={`thread-count-${item.id}`}>
                <i className="far fa-file-alt"></i>{" "}
                {this.state.threadCounts.length === this.state.boards.length ? (
                  this.getCount(item.id)
                ) : (
                  <img
                    src="/images/loader/wave-loader.svg"
                    alt="wave-loader"
                    id="wave-loader-gif"
                  />
                )}
              </p>
            </span>
          </Link>
        </li>
      );
    });
  };
  componentDidMount() {
    window.scroll(0, 0);
    const { forum_cat } = this.props.match.params;
    apiServices
      .getForumSectionTitles()
      .then(forumTitles => {
        const title = forumTitles.filter(
          title => title.id.toString() === forum_cat
        );
        this.setState({
          forumTitle: title[0].name
        });
      })
      .then(() => {
        apiServices
          .getForumBoards(forum_cat)
          .then(boards => {
            this.setState({
              boards: boards
            });
          })
          .then(() => {
            var counts = [];
            this.state.boards.forEach(board => {
              apiServices
                .getNumOfThreads(board.id)
                .then(numOfThreads => {
                  counts.push({
                    board_id: board.id,
                    count: numOfThreads[0].count
                  });
                })
                .then(() => {
                  this.setState({
                    threadCounts: counts
                  });
                });
            });
          });
      });
  }

  render() {
    return (
      <section className="forum-cat-sections-container">
        <div className="forum-cat-sections-content">
          <header>
            <p>{this.state.forumTitle}</p>
          </header>
          {this.state.boards ? <ul>{this.makeBoards()}</ul> : <p>Loading...</p>}
        </div>
      </section>
    );
  }
}
