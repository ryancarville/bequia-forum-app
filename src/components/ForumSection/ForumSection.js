import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import Truncate from 'react-truncate';
import STORE from '../../STORE/store';
import CreatePostButton from '../CreatePostButton/CreatePostButton';
import './ForumSection.css';
import TokenServices from '../../services/TokenServices';
import formatDate from '../../helpers/formatDate';
import like from '../Icons/like';
import comment from '../Icons/comment';

class ForumSection extends Component {
	constructor(props) {
		super(props);
		this.state = {
			posts: STORE.posts
		};
	}

	getPosts = () => {
		const forumId = this.props.match.params.forumId;
		const forumSectionPosts = this.state.posts.filter(
			p => p.forumId.toString() === forumId
		);

		const visiblePosts = forumSectionPosts.map(p => {
			console.log('log here', p);
			const numOfComments = STORE.comments.filter(
				comment => comment.postId === p.id
			).length;
			return (
				<li key={p.id}>
					<Link to={`/messageBoard/${forumId}/${p.id}`}>{p.title}</Link>
					<br />
					<Truncate
						lines={1}
						ellipsis={
							<span>
								...
								<Link to={`/messageBoard/${forumId}/${p.id}`}>Read more</Link>
							</span>
						}>
						{p.content}
					</Truncate>
					<span className='postInfo'>
						<p>Posted By: {p.author || p.contact.name}</p>
						<p>Posted On: {formatDate(p.date)}</p>
						<span className='post-icons'>
							<p>
								{like}
								{'   '}
								{p.likes}
							</p>
							<p>
								{comment}
								{'   '}
								{numOfComments}
							</p>
						</span>
					</span>
				</li>
			);
		});

		return forumSectionPosts.length === 0 ? (
			<p>There are currently no posts in this section. Be the first one!</p>
		) : (
			visiblePosts
		);
	};

	render() {
		const forumId = this.props.match.params.forumId;

		return (
			<div className='forum-section-container'>
				<h3>{this.props.location.state.forum.title}</h3>
				<span>
					{TokenServices.getAuthToken() ? (
						<CreatePostButton forumId={forumId} />
					) : null}
				</span>
				<div className='forum-section-content'>
					<ul>{this.getPosts()}</ul>
				</div>
			</div>
		);
	}
}
export default withRouter(ForumSection);
