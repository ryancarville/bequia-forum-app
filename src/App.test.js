import React from 'react';
import Router from '../src/Router/Router';
import Footer from '../src/components/Footer/Footer';
import Nav from '../src/components/Nav/Nav';
import Forum from '../src/components/Forum/Forum';
import ShallowRenderer from 'react-test-renderer/shallow';
import ForumCatSections from './components/ForumCatSections/ForumCatSections';
import ForumSection from './components/ForumSection/ForumSection';
import ForumContext from '../src/ForumContext';
import Jobs from './components/Jobs/Jobs';

const renderer = new ShallowRenderer();
const tree = renderer.getRenderOutput();
it('renders user shallow correctly', () => {
	renderer.render(<Nav />, <Router />, <Footer />);
	expect(tree).toMatchSnapshot();
});

it('renders forum correctly', () => {
	renderer.render(<Forum />);
	expect(tree).toMatchSnapshot();
});

it('renders forum catagories correctly', () => {
	renderer.render(<ForumCatSections />);
	expect(tree).toMatchSnapshot();
});

it('renders forum section correctly', () => {
	renderer.render(<ForumSection />);
	expect(tree).toMatchSnapshot();
});

it('renders jobs correctly', () => {
	renderer.render(<Jobs context={ForumContext} />);
	expect(tree).toMatchSnapshot();
});
