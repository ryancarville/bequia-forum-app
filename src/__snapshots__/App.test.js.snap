// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Dashboard components renders dashboard correctly 1`] = `null`;

exports[`All Directory Components renders directory correctly 1`] = `null`;

exports[`All Forum Components renders New Posts correctly 1`] = `null`;

exports[`All Forum Components renders forum catagories correctly 1`] = `null`;

exports[`All Forum Components renders forum correctly 1`] = `null`;

exports[`All Forum Components renders forum section correctly 1`] = `null`;

exports[`All Jobs Components renders jobs cats correctly 1`] = `null`;

exports[`All Jobs Components renders jobs correctly 1`] = `null`;

exports[`All Jobs Components renders jobs page cats correctly 1`] = `null`;

exports[`All Market Place Components renders Market Place Section Correctly 1`] = `null`;

exports[`All Market Place Components renders Market Place correctly 1`] = `null`;

exports[`All Market Place Components renders market Place listing correctly 1`] = `null`;

exports[`All Rental Components renders rental cats correctly 1`] = `null`;

exports[`All Rental Components renders rental page correctly 1`] = `null`;

exports[`All Rental Components renders rentals correctly 1`] = `null`;

exports[`App Component renders Nav correctly 1`] = `null`;

exports[`App Component renders Router correctly 1`] = `null`;

exports[`App Component renders footer correctly 1`] = `null`;
